tests/ottest/fixtures/blockchain/Restart.cpp

```diff
@@ -0,0 +1,264 @@
+
+    auto future = network.SendToAddress(
+        sender.nym_id_, address, coin_to_send_, memo_outgoing);
+
+    // We need to confirm send transaction, so it is available after restore
+    mineTransaction(sender, future.get().second);
+}
+
+void Restart_fixture::collectOutputs(
+    const User& user_bob,
+    const User& user_alice,
+    Vector<UTXO>& all_bobs_outputs,
+    Vector<UTXO>& all_alice_outputs,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_bob,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_alice)
+{
+    const auto& bobs_network =
+        user_bob.api_->Network().Blockchain().GetChain(test_chain_);
+    const auto& bobs_wallet = bobs_network.Wallet();
+
+    const auto& alice_network =
+        user_alice.api_->Network().Blockchain().GetChain(test_chain_);
+    const auto& alice_wallet = alice_network.Wallet();
+
+    all_bobs_outputs = bobs_wallet.GetOutputs(TxoState::All);
+    all_alice_outputs = alice_wallet.GetOutputs(TxoState::All);
+
+    auto all_types = {
+        TxoState::Immature,
+        TxoState::ConfirmedSpend,
+        TxoState::UnconfirmedSpend,
+        TxoState::ConfirmedNew,
+        TxoState::UnconfirmedNew,
+        TxoState::OrphanedSpend,
+        TxoState::OrphanedNew};
+    for (const auto& type : all_types) {
+        number_of_outputs_per_type_bob[type] =
+            bobs_wallet.GetOutputs(type).size();
+        number_of_outputs_per_type_alice[type] =
+            alice_wallet.GetOutputs(type).size();
+    }
+}
+
+void Restart_fixture::collectOutputsAsSet(
+    const User& user_bob,
+    const User& user_alice,
+    OutputsSet& all_bobs_outputs,
+    OutputsSet& all_alice_outputs,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_bob,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_alice)
+{
+    // Wait for outputs to process
+    sleep(20);
+
+    Vector<UTXO> bobs_outputs;
+    Vector<UTXO> alice_outputs;
+
+    collectOutputs(
+        user_bob,
+        user_alice,
+        bobs_outputs,
+        alice_outputs,
+        number_of_outputs_per_type_bob,
+        number_of_outputs_per_type_alice);
+
+    EXPECT_EQ(bobs_outputs.size(), bobs_outputs.size());
+    EXPECT_EQ(alice_outputs.size(), alice_outputs.size());
+
+    for (std::size_t i = 0; i < bobs_outputs.size(); ++i)
+        all_bobs_outputs.insert(std::move(bobs_outputs[i]));
+
+    for (std::size_t i = 0; i < alice_outputs.size(); ++i)
+        all_alice_outputs.insert(std::move(alice_outputs[i]));
+}
+
+std::pair<ot::Amount, ot::Amount> Restart_fixture::collectFeeRate(
+    const User& user_bob,
+    const User& user_alice)
+{
+    const auto& bobs_network =
+        user_bob.api_->Network().Blockchain().GetChain(test_chain_);
+
+    const auto& alice_network =
+        user_alice.api_->Network().Blockchain().GetChain(test_chain_);
+
+    return std::make_pair(
+        bobs_network.Internal().FeeRate(), alice_network.Internal().FeeRate());
+}
```

Comment:
dealing with fees is tricky. Feerate does not provide us much information for balance calculations. Feerate is just the estimate about the amount of gas/satoshi/fee we need to pay per kB of data stored on the blockchain. In order to reliably calculate the fee which was actually paid to submit the transaction we should rather parse the transaction, get all Inputs and their Values, get all Outputs and their Values, substract `InputsTotal - OutputsTotal` and the result is the fee paid.